ch5/exercise5-9/expand.go

```diff
@@ -0,0 +1,42 @@
+package main 
+
+
+import (
+	"strings"
+	"fmt"
+)
+const (
+	FOO = "$foo"
+)
+
+func main() {
+	fmt.Println(expand("$foo $foo $fo", func(x string)string {
+		return "random"
```

Comment:
I think it would be more in the spirit of the exercise to have `f(x)` use x somehow (which also means considering more than just `$foo`  For instance:

```go
func double(s string) string {
    return s + s
}
```
```go
var fourLetterWords = map[string]bool{"darn": true, "dang": true, "heck": true}

func censor(s string) string {
        if _, ok := fourLetterWords[s]; ok {
                return "****"
        }
        return s
}
```

Reply:
Ok, I will update it to use x